### ISSUE TITLE
remoteproc_virtio_notify return status not used

### ISSUE THREAD

**Issue by tammyleino** [OpenAMP/open-amp#343]
The remoteproc_virtio_notify function is used in functions defined in remoteproc_virtio.c as rpvdev->notify.  The implementation is ultimately defined by the BSP. This function returns the status of the operation but the status is not checked by any of the calling routines.  Should the return status of remoteproc_virtio_notify be changed to void?

**Comment by edmooring** [comment]
I hate losing potential error information, but I also can't see what the caller could sanely do in the face of an error. I am not opposed to changing the return status. However, this is an API change that will break existing code, so we have to be careful. 

@arnopo Do you think we could do this with a CMake setting?

**Comment by arnopo** [comment]
I can see 2 issues: 

1.  The ` rpvdev->notify`  in  remoteproc_virtio.c can be null, leading to crash

=>   `rpvdev->notify ==  0` is valid if user doesn't use a mailbox to notify the remote processor. The call should be protected by `if (rpvdev->notify)`

2.  The status is not checked.  

=> what to do in case of error? This depends on the virtio backend implementation. The remoteproc_virtio is only one.
So change the API seems not a solution.
I would be in favor of  testing the status in remoteproc_virtio.c and printing a warning.

**Comment by tammyleino** [comment]
@arnopo Yes, the function pointer should definitely be checked before using it.  I will open a PR eventually for this and possibly other remoteproc op function pointers.  I think logging the error will be fine for OpenAMP in general, but I think in my particular case, I need to not allow an error to be returned from the driver's notify routine.  My driver is just triggering an inter-core interrupt via a register, which won't fail.  The potential fail case is if the incoming remoteproc pointer is null.  This should have been caught at the API level before getting all the way to the driver.

**Comment by atulakella** [comment]
Should I send a PR for this?
Plan is to null-check `rpvdev->notify` before calling it (a null notify is valid, just means no mailbox is configured, so return early), and log a warning via `metal_log` on nonzero return instead of changing the exported return type, per the discussion above. Will add a host-side unit test covering both the null and nonzero-return notify paths.

**Comment by arnopo** [comment]
> Should I send a PR for this? Plan is to null-check `rpvdev->notify` before calling it (a null notify is valid, just means no mailbox is configured, so return early), and log a warning via `metal_log` on nonzero return instead of changing the exported return type, per the discussion above. Will add a host-side unit test covering both the null and nonzero-return notify paths.

yes feel free, thanks!
